### PR TITLE
Don't set output variable and Log warning if helm command output leng…

### PR DESCRIPTION
**Pull request by wasfree**
**Task name**: HelmDeployV0

**Description**: Output variable should not be set if Helm command output exceeded the maximum supported variable length.

**Documentation changes required:** (Y/N) N

**Added unit tests:** (Y/N) N

**Attached related issue:** (Y/N) Was also done for kubectl and docker task see #9379

**Checklist**:
- [ x ] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [ x ] Checked that applied changes work as expected

### CODE REVIEW

File: Tasks/HelmDeployV0/src/helm.ts [hunk 1]
```diff
@@ -157,9 +158,15 @@ function runHelm(helmCli: helmcli, command: string, kubectlCli: kubernetescli, f
 
     const execResult = helmCli.execHelmCommand();
     tl.setVariable('helmExitCode', execResult.code.toString());
-    if (execResult.stdout) {
-        tl.setVariable('helmOutput', execResult.stdout);
+
+    var commandOutputLength = execResult.length;
```
Comment (evertonmc) [inline Tasks/HelmDeployV0/src/helm.ts:162]: `execResult.stdout.length`

**Review by evertonmc — CHANGES_REQUESTED** [review]
`Property 'length' does not exist on type 'IExecSyncResult'.`

**Review by evertonmc — APPROVED** [review]
LGTM

File: Tasks/HelmDeployV0/src/helm.ts [hunk 2]
```diff
@@ -157,9 +158,15 @@ function runHelm(helmCli: helmcli, command: string, kubectlCli: kubernetescli, f
 
     const execResult = helmCli.execHelmCommand();
     tl.setVariable('helmExitCode', execResult.code.toString());
-    if (execResult.stdout) {
-        tl.setVariable('helmOutput', execResult.stdout);
+
+    var commandOutputLength = execResult.stdout.length;
```
Comment (shigupt202) [inline Tasks/HelmDeployV0/src/helm.ts:162]: The `length` property of stdout should be accessed after the null check for `execResult.stdout`. You can put the new logic inside the already existing if condition.

File: Tasks/HelmDeployV0/task.json [hunk 3]
```diff
@@ -14,7 +14,7 @@
     "version": {
         "Major": 0,
         "Minor": 178,
```
Comment (shigupt202) [inline Tasks/HelmDeployV0/task.json:16]: Could you please update the Minor to 179, and Patch to 0 as this is a new sprint.

File: Tasks/HelmDeployV0/src/helm.ts (hunk 2 above)
Comment (wasfree) [reply Tasks/HelmDeployV0/src/helm.ts]: The `length` property of stdout is now inside condition. Thanks.

File: Tasks/HelmDeployV0/task.json (hunk 3 above)
Comment (wasfree) [reply Tasks/HelmDeployV0/task.json]: Version was bumped minor to 179 and patch to 0

**Comment by shigupt202** [comment]
@wasfree Thanks for the changes. Could you please confirm what you're seeing currently when the command output exceeds the maximum supported variable length, if you have a repro (Does it give an exception, or truncates the content)? Just want to be sure that this change won't break back compat.

**Comment by wasfree** [comment]
> @wasfree Thanks for the changes. Could you please confirm what you're seeing currently when the command output exceeds the maximum supported variable length, if you have a repro (Does it give an exception, or truncates the content)? Just want to be sure that this change won't break back compat.

@shigupt202 installed the new version of helm task and it works as expected. At the task we can see all output now and at the end users will receive this warning if length exceeded:
```
##[warning]Output variable not set as Helm command output exceeded the maximum supported length. Output length: 378983, Maximum supported length: 32766
```

**Comment by shigupt202** [comment]
> > @wasfree Thanks for the changes. Could you please confirm what you're seeing currently when the command output exceeds the maximum supported variable length, if you have a repro (Does it give an exception, or truncates the content)? Just want to be sure that this change won't break back compat.
> 
> @shigupt202 installed the new version of helm task and it works as expected. At the task we can see all output now and at the end users will receive this warning if length exceeded:
> 
> ```
> ##[warning]Output variable not set as Helm command output exceeded the maximum supported length. Output length: 378983, Maximum supported length: 32766
> ```

I meant without this change, what is the behaviour? If currently (without this change) the task is setting the truncated output variable, then it would mean this change is breaking back compat in a way because now we are not setting the variable to anything and this could break some of the pipelines which were relying on this output variable.

**Comment by wasfree** [comment]
> > > @wasfree Thanks for the changes. Could you please confirm what you're seeing currently when the command output exceeds the maximum supported variable length, if you have a repro (Does it give an exception, or truncates the content)? Just want to be sure that this change won't break back compat.
> > 
> > 
> > @shigupt202 installed the new version of helm task and it works as expected. At the task we can see all output now and at the end users will receive this warning if length exceeded:
> > ```
> > ##[warning]Output variable not set as Helm command output exceeded the maximum supported length. Output length: 378983, Maximum supported length: 32766
> > ```
> 
> I meant without this change, what is the behaviour? If currently (without this change) the task is setting the truncated output variable, then it would mean this change is breaking back compat in a way because now we are not setting the variable to anything and this could break some of the pipelines which were relying on this output variable.



> > > @wasfree Thanks for the changes. Could you please confirm what you're seeing currently when the command output exceeds the maximum supported variable length, if you have a repro (Does it give an exception, or truncates the content)? Just want to be sure that this change won't break back compat.
> > 
> > 
> > @shigupt202 installed the new version of helm task and it works as expected. At the task we can see all output now and at the end users will receive this warning if length exceeded:
> > ```
> > ##[warning]Output variable not set as Helm command output exceeded the maximum supported length. Output length: 378983, Maximum supported length: 32766
> > ```
> 
> I meant without this change, what is the behaviour? If currently (without this change) the task is setting the truncated output variable, then it would mean this change is breaking back compat in a way because now we are not setting the variable to anything and this could break some of the pipelines which were relying on this output variable.

Without this change the task will fail with output "tasks failing reason: Argument list too long", so will never complete.

**Review by shigupt202 — APPROVED** [review]

**Pull request merged**